source/javascripts/components/StacksAndMachine/DefaultStackAndMachine.tsx

```diff
@@ -74,13 +76,21 @@ const getDeprecatedMachinesProps = (
 };
 
 const DefaultStackAndMachine = () => {
+  const { data } = useStacksAndMachines();
   const { stackId, machineTypeId, stackRollbackVersion } = useDefaultStackAndMachine();
   const updateStacksAndMachinesMeta = useBitriseYmlStore((s) => s.updateStacksAndMachinesMeta);
 
   const updateDefaultMeta = (stack: string, machine_type_id: string, stack_rollback_version: string) => {
+    const finalMachineTypeId = StackAndMachineService.selectFinalMachineTypeId({
+      ...data,
+      defaultMachineTypeId: machineTypeId,
+      selectedStackId: stack,
+      selectedMachineTypeId: machine_type_id,
```

Comment:
Some better naming would be appreciated, especially `machine_type_id` could use a better name.